src/game_server.rs

```diff
@@ -66,7 +66,7 @@ impl GameServer {
     }
 
     pub fn on_new_action(&mut self, client: usize, action: Action) {
-        self.world.apply_action(&action);
+        self.world.apply_action_no_render(&action);
```

Comment:
This is not good. You will see my comments in `world.rs` that explains how to fix this.